Libraries/Text/RCTTextView.m

```diff
@@ -336,6 +337,22 @@ - (void)setContentInset:(UIEdgeInsets)contentInset
 
 - (BOOL)textView:(RCTUITextView *)textView shouldChangeTextInRange:(NSRange)range replacementText:(NSString *)text
 {
+  if (textView.textContainer.maximumNumberOfLines > 0) {
```

Comment:
This method is already pretty huge. Can you move this to where we evaluate maxLength?